File: src/main/kotlin/com/skywalker/core/exception/ServiceException.java
package com.skywalker.core.exception;

/**
 * product exception class
 * 
 * @author chenkb
 *
 */
public class ServiceException extends Exception
{
	/**
	 * uid 
	 */
	private static final long serialVersionUID = 1L;
	
	protected String errorKey;
    protected Object value1;
    
    /**
     * construct an exception with null as the detailed message
     *
     */
    public ServiceException()
    {
        super();
    }
    
    /**
     * construct an exception with the specified error message
     * 
     * @param message
     */
    public ServiceException(String errorKey, String message)
    {
        super(message);
        this.errorKey = errorKey;
    }
    
    /**
     * construct an exception with error message and root cause
     * 
     * @param message
     * @param cause
     */
    public ServiceException(String message, Throwable cause)
    {
        super(message, cause); 
    }
    
    /**
     * construct an exception with errorkey
     * 
     * @param errorKey
     */
    public ServiceException(String errorKey)
    {        
        this.errorKey = errorKey;
    }
    
    /**
     * construct an exception with the specified error message
     * 
     * @param message
     */
    public ServiceException(String errorKey, String errorCode, String message)
    {
        super(message);
        this.errorKey = errorKey;
    }

    public ServiceException(String errorKey, String errorCode, String message, Object value1)
    {
        super(message);
        this.errorKey = errorKey;
        this.value1 = value1;
    }
    /**
     * the getter method for error key
     * 
     * @return
     */
    public String getErrorKey()
    {
    	return this.errorKey;
    }  

	/**
	 * @return Returns the value.
	 */
	public Object getValue1()
	{
		return value1;
	}

	
}

File: src/main/kotlin/com/skywalker/core/exception/ServiceException.kt
package com.skywalker.core.exception

import org.springframework.validation.FieldError

/**
 * 异常类
 */
class ServiceException : Exception {

    var errorKey: String = "-1000"
    var value: String = "系统错误"

    constructor() : super()

    constructor(errorKey: String, message: String) : super(message) {
        this.errorKey = errorKey
        this.value = message
    }
    constructor(errorKey: String, message: String,e:Exception) : super(message) {
        e.printStackTrace()
        this.errorKey = errorKey
        this.value = message
    }
    constructor(errorKey: String, fieldErrors: List<FieldError>){
        this.errorKey = errorKey
        val sb = StringBuffer()
        for (error in fieldErrors) {
            sb.append(error.defaultMessage).append(";")
        }
        this.value = sb.toString()
    }

}
